Docker build from source fails out-of-the-box: .dockerignore excludes mcp/, cli/, scripts/ that the Dockerfile needs

## Summary

Building the Docker image from source (`docker compose up --build` / `docker build .`) fails out-of-the-box on `main` (and on the `v5.0.0` / `v5.0.1` tags). The root `.dockerignore` excludes directories that the `Dockerfile` then tries to `COPY`, and the `deps` stage runs `pnpm install` without `--ignore-scripts`, so the `prepare` lifecycle hook fails. Self-hosters following the Docker path in `docs/guides/SELF_HOST.md` cannot build the image without patching these two files.

## Environment

- Repo state: `main` (verified the same on tags `v5.0.0` and `v5.0.1`)
- Build: `docker compose up --build` using the root `Dockerfile`
- Node 22 / pnpm 9.15.4 (as pinned in the Dockerfile `deps` stage)

## Failure 1 — `.dockerignore` excludes `mcp/` and `cli/` but the Dockerfile COPYs them

`.dockerignore` (lines ~48-51):

```
# Development tools
cli/
mcp/
scripts/
tasks/
```

But the `Dockerfile` `deps` stage copies the per-workspace manifests:

```dockerfile
COPY cli/package.json ./cli/
COPY mcp/package.json ./mcp/
```

Because `cli/` and `mcp/` are dockerignored, those paths are not in the build context and the build fails:

```
failed to solve: failed to compute cache key: failed to calculate checksum of ref ...: "/mcp/package.json": not found
```

## Failure 2 — `deps` stage `pnpm install` triggers the `prepare` hook, which needs `scripts/`

After making `cli/`/`mcp/` available, the next failure is in the `deps` stage:

```
. prepare: Error: Cannot find module '/app/scripts/prepare-husky.mjs'
[ELIFECYCLE] Command failed with exit code 1.
ERROR: process "/bin/sh -c pnpm install --frozen-lockfile" did not complete successfully: exit code: 1
```

Root cause: the root `package.json` has `"prepare": "node scripts/prepare-husky.mjs"`. The `deps` stage runs:

```dockerfile
RUN pnpm install --frozen-lockfile
```

without `--ignore-scripts`, so pnpm runs the `prepare` hook. But the `deps` stage only copies the workspace `package.json` files (not `scripts/`), and `scripts/` is also dockerignored, so `scripts/prepare-husky.mjs` does not exist at that point. Note the `production` stage already does this correctly with `pnpm install --frozen-lockfile --prod --ignore-scripts` — the `deps` stage is just inconsistent.

## Proposed fix

Two small, consistent changes:

**1. `.dockerignore`** — stop excluding the directories the Dockerfile needs. `cli/` and `mcp/` are COPYed; `scripts/` is needed by the `prepare` hook. (`tasks/` can stay excluded — it is runtime data from a volume.)

```diff
 # Development tools
-cli/
-mcp/
-scripts/
 tasks/
```

**2. `Dockerfile`** (`deps` stage) — make `scripts/` available and skip husky during the image build, so the `prepare` hook is a no-op:

```dockerfile
COPY cli/package.json ./cli/
COPY mcp/package.json ./mcp/
COPY scripts/ ./scripts/
ENV HUSKY=0
RUN pnpm install --frozen-lockfile
```

`scripts/prepare-husky.mjs` already short-circuits when `HUSKY=0` (or `NODE_ENV=production` / `CI=true`), so the hook exits cleanly once the file is present. Keeping the install (rather than adding `--ignore-scripts` here) preserves native dependency builds such as `better-sqlite3`.

With both changes applied, `docker compose up --build` builds cleanly, the container boots healthy on `GET /health`, and both `VERITAS_STORAGE=file` and `VERITAS_STORAGE=sqlite` modes work (verified against a real v3 -> v5 file-to-SQLite migration of ~265 tasks via `POST /api/v1/sqlite/migration/run`).

Happy to open a PR with these two changes if useful.

--- a/.dockerignore
+++ b/.dockerignore
@@ -45,9 +45,6 @@ LICENSE
 coverage
 
 # Development tools
-cli/
-mcp/
-scripts/
 tasks/
 eslint.config.js
 .prettierrc

--- a/Dockerfile
+++ b/Dockerfile
@@ -16,7 +16,7 @@
 # ---------------------------------------------------------------------------
 FROM node:22-alpine AS deps
 
-RUN corepack enable && corepack prepare pnpm@9.15.4 --activate
+RUN corepack enable && corepack prepare pnpm@11.1.1 --activate
 
 WORKDIR /app
 
@@ -27,8 +27,10 @@ COPY server/package.json ./server/
 COPY web/package.json ./web/
 COPY cli/package.json ./cli/
 COPY mcp/package.json ./mcp/
+COPY scripts/ ./scripts/
 
 # Install all dependencies (dev + prod) for building
+ENV HUSKY=0
 RUN pnpm install --frozen-lockfile
 
 # ---------------------------------------------------------------------------
@@ -65,7 +67,7 @@ RUN pnpm --filter @veritas-kanban/server build
 # ---------------------------------------------------------------------------
 FROM node:22-alpine AS production
 
-RUN corepack enable && corepack prepare pnpm@9.15.4 --activate
+RUN corepack enable && corepack prepare pnpm@11.1.1 --activate
 
 # Security: run as non-root
 RUN addgroup -g 1001 -S nodejs && \

--- a/README.md
+++ b/README.md
@@ -335,7 +335,7 @@ Tasks are markdown files. Settings are JSON. Workflows are YAML. No database, no
 | **Git**             | simple-git, worktree management      | —                                |
 | **Testing**         | Playwright (E2E), Vitest (unit)      | Playwright 1.58, Vitest 4        |
 | **Runtime**         | Node.js                              | 22+                              |
-| **Package Manager** | pnpm                                 | 9+                               |
+| **Package Manager** | pnpm                                 | 11.1.1+                          |
 
 ---
 

--- a/docs/DEPLOYMENT.md
+++ b/docs/DEPLOYMENT.md
@@ -226,13 +226,13 @@ If you need to debug inside a container, use `docker exec` to inspect — don't 
 | Requirement | Version |
 | ----------- | ------- |
 | Node.js     | 22.0.0+ |
-| pnpm        | 9.0.0+  |
+| pnpm        | 11.1.1+ |
 
 Install pnpm if not present:
 
 ```bash
 corepack enable
-corepack prepare pnpm@9.15.4 --activate
+corepack prepare pnpm@11.1.1 --activate
 ```
 
 ### Build Steps

--- a/docs/GETTING-STARTED.md
+++ b/docs/GETTING-STARTED.md
@@ -28,12 +28,12 @@ A working board is not the same as agent-ready or external wake/delivery-ready. 
 
 ## Prerequisites (30 seconds)
 
-| What              | Command            | Notes                                              |
-| ----------------- | ------------------ | -------------------------------------------------- |
-| Node.js           | `node -v`          | Requires **22+**. Install via Volta/nvm if older.  |
-| pnpm              | `pnpm -v`          | Requires **9+**. `npm install -g pnpm` if missing. |
-| Git               | `git --version`    | Any current version works.                         |
-| (Optional) Docker | `docker --version` | Needed only if you prefer containers.              |
+| What              | Command            | Notes                                                                   |
+| ----------------- | ------------------ | ----------------------------------------------------------------------- |
+| Node.js           | `node -v`          | Requires **22+**. Install via Volta/nvm if older.                       |
+| pnpm              | `pnpm -v`          | Requires **11.1.1+**. Prefer `corepack prepare pnpm@11.1.1 --activate`. |
+| Git               | `git --version`    | Any current version works.                                              |
+| (Optional) Docker | `docker --version` | Needed only if you prefer containers.                                   |
 
 That's it. No database, no extra services.
 

--- a/docs/guides/SELF_HOST.md
+++ b/docs/guides/SELF_HOST.md
@@ -33,14 +33,14 @@ This guide walks you through every self-hosting scenario — from running locall
 | Requirement | Version | Install                                                      |
 | ----------- | ------- | ------------------------------------------------------------ |
 | Node.js     | 22.0.0+ | https://nodejs.org or `nvm install 22`                       |
-| pnpm        | 9.0.0+  | `corepack enable && corepack prepare pnpm@9.15.4 --activate` |
+| pnpm        | 11.1.1+ | `corepack enable && corepack prepare pnpm@11.1.1 --activate` |
 | Git         | any     | https://git-scm.com                                          |
 
 Verify:
 
 ```bash
 node --version   # v22.x.x
-pnpm --version   # 9.x.x
+pnpm --version   # 11.x.x
 ```
 
 ---

